app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  include DeviseTokenAuth::Concerns::User
+
+  has_many :targets
+
+  validates :full_name, presence: true
```

Comment:
maybe you can control if the email is unique and check the presence of other attributes. though i don't remember if devise does it for you.

Reply:
Also, doesn't a user have a gender? I don't know if the attribute was going to be implemented down the road. If the user has a gender, its presence and inclusion in the valid genders should be validated.

Reply:
Thanks ! Devise already check the presence and uniqueness of email.